metadata-ingestion/src/datahub/ingestion/source/sql/bigquery.py

```diff
@@ -398,28 +397,42 @@ def _get_bigquery_log_entries(
         template: str,
     ) -> Union[Iterable[AuditLogEntry], Iterable[BigQueryAuditMetadata]]:
         # Add a buffer to start and end time to account for delays in logging events.
+        start_time = (self.config.start_time - self.config.max_query_duration).strftime(
+            BQ_DATETIME_FORMAT
+        )
+        self.report.start_time = start_time
+
+        end_time = (self.config.end_time + self.config.max_query_duration).strftime(
+            BQ_DATETIME_FORMAT
+        )
+        self.report.end_time = end_time
+
         filter = template.format(
-            start_time=(
-                self.config.start_time - self.config.max_query_duration
-            ).strftime(BQ_DATETIME_FORMAT),
-            end_time=(self.config.end_time + self.config.max_query_duration).strftime(
-                BQ_DATETIME_FORMAT
-            ),
+            start_time=start_time,
+            end_time=end_time,
         )
 
         assert self.config.log_page_size is not None
 
-        logger.info("Start loading log entries from BigQuery")
+        logger.info(
+            f"Start loading log entries from BigQuery start_time={start_time} and end_time={end_time}"
+        )
         for client in clients:
             entries = client.list_entries(
                 filter_=filter, page_size=self.config.log_page_size
             )
             item = 0
             for entry in entries:
                 item = item + 1
+
+                # In loop intentionally so in case of failures we do get numbers
+                self.report.num_total_log_entries = item
```

Comment:
1. Should this be a per-client counter instead of a global one? Otherwise, initialize item(probably renamed suitably) outside the loop?
2. What failures are expected here? If we have unhandled failures in this code, please use try...except...finally and update in the finally clause.

Reply:
will fix